grails-app/services/au/org/emii/gogoduck/job/NotificationService.groovy

```diff
@@ -85,4 +86,12 @@ class NotificationService {
     def methodMissing(String name, args) {
         log.debug("sending email, ${name}: ${args}")
     }
+
+    def getJobUrl(job) {
+        return grailsLinkGenerator.link(controller: 'job', action: 'show', id: job.uuid, absolute: true)
```

Comment:
This the core of my latest commit.  Grails can take what's in `UrlMappings` and a call to `link` like this to generate correct URLs for us.